sympy/integrals/manualintegrate.py

```diff
@@ -1528,6 +1535,23 @@ def eval_reciprocal_sqrt_quadratic(a, b, c, integrand, x):
     return log(2*sqrt(c)*sqrt(a+b*x+c*x**2)+b+2*c*x)/sqrt(c)
 
 
+@evaluates(SqrtQuadraticDenomRule)
+def eval_sqrt_quadratic_denom(a, b, c, coeffs: list[Expr], integrand, x):
+    result_coeffs = []
+    coeffs = coeffs.copy()
+    for i in range(len(coeffs)-2):
+        n = len(coeffs)-1-i
+        coeff = coeffs[i]/(c*n)
+        result_coeffs.append(coeff)
+        coeffs[i+1] -= (2*n-1)*b/2*coeff
+        coeffs[i+2] -= (n-1)*a*coeff
+    d, e = coeffs[-1], coeffs[-2]
+    s = sqrt(a+b*x+c*x**2)
+    constant = d-b*e/(2*c)
```

Comment:
I guess there should be a Piecewise to handle `c=0`.

Reply:
This is guarded at https://github.com/sympy/sympy/blob/78bbfa804d00899aa92d61eb4f012d6f31d928c3/sympy/integrals/manualintegrate.py#L837 so there won't be divided-by-zero at this place.
However, nothing can prevent a user to use a Symbol `c` here the integration result, and substitute it with 0 in the integration result. So the question is still "whether we want to handle every possible degenerated case".

Reply:
Generally integrate() does return a Piecewise for the cases where divided symbolic constants can be 0. There are options to remove the Piecewise if you don't want it, although they currently only work for meijerint (that's a separate issue).

Reply:
Now a `PiecewiseRule` is generated before calling this function, so substituting `c` with 0 is safe.